File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/I.Степень_четырёх/solution.py
import sys


def solution(n: int) -> None:
    """
    Time Complexity O(log4 N)
    Memory Complexity O(1)
    """
    cur_n: int = 1
    upper_bound: int = 10_001
    is_valid: bool = False
    while cur_n < upper_bound:
        if cur_n == n:
            is_valid = True
            break
        cur_n *= 4
    sys.stdout.write('True' if is_valid else 'False')


if __name__ == '__main__':
    n: int = int(sys.stdin.readline().rstrip())
    solution(n)

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/I.Степень_четырёх/solution.cpp
#include <iostream>


void solution(int n) {
    /*
    Time Complexity O(log4 N)
    Memory Complexity O(1)
    */
    int mod = 4;
    int curN = 1;
    int upperBound = 10001;
    bool isValid = false;
    while (curN < upperBound) {
        if (curN == n) {
            isValid = true;
            break;
        }
        curN *= 4;
    }
    std::cout << (isValid? "True" : "False") << "\n";
};


int main() {
    int n;
    std::cin >> n;
    solution(n);
}